modulemd/modulemd-defaults-v1.c

```diff
@@ -1322,45 +1316,43 @@ modulemd_defaults_v1_merge (const gchar *module_name,
       merged_default_stream =
         g_hash_table_lookup (merged->intent_default_streams, intent_name);
 
-      if (!merged_default_stream)
+      /* If there is no new default stream, just jump to the next item */
+      if (!intent_default_stream)
+        continue;
+
+      else if (!merged_default_stream && intent_default_stream)
         {
           /* New entry, just add it */
           g_hash_table_insert (merged->intent_default_streams,
                                g_strdup (intent_name),
                                g_strdup (intent_default_stream));
-          continue;
         }
 
-      if (!g_str_equal (intent_default_stream, merged_default_stream))
+      else if (!g_str_equal (intent_default_stream, merged_default_stream))
         {
-          g_set_error (error,
-                       MODULEMD_ERROR,
-                       MODULEMD_ERROR_VALIDATE,
-                       "Profile default stream mismatch in intents: %s != %s",
-                       intent_default_stream,
-                       merged_default_stream);
-          return NULL;
+          if (from_modified > into_modified)
+            {
+              g_hash_table_replace (merged->intent_default_streams,
+                                    g_strdup (intent_name),
+                                    g_strdup (intent_default_stream));
+            }
+          else if (into_modified == from_modified)
+            {
+              g_set_error (
+                error,
+                MODULEMD_ERROR,
+                MODULEMD_ERROR_VALIDATE,
+                "Profile default stream mismatch in intents: %s != %s",
+                intent_default_stream,
+                merged_default_stream);
+              return NULL;
+            }
         }
     }
 
 
   /* Merge intent default profile values */
 
-  /* First copy 'into' to 'merged' */
-  g_hash_table_iter_init (&iter, into->intent_default_profiles);
-  while (g_hash_table_iter_next (&iter, &key, &value))
-    {
-      intent_name = (gchar *)key;
-      intent_profiles = (GHashTable *)value;
-
-      intent_profile_defaults =
-        modulemd_defaults_v1_copy_intent_profiles (intent_profiles);
-
-      g_hash_table_insert (merged->intent_default_profiles,
-                           g_strdup (intent_name),
-                           g_steal_pointer (&intent_profile_defaults));
-    }
-
   /* Now copy 'from" into merged, checking for conflicts */
```

Comment:
```suggestion
   /* Now copy 'from' into merged, checking for conflicts */
```
That was an interesting mixture of quotes!